apps/github-api-integration/README.md

```diff
@@ -0,0 +1,212 @@
+
+```js
+validate(req, res, next) {
+  const name = req.form.values.name;
+
+  // Define an instance of the Model
+  // with attributes and config (uri)
+  const model = new Model({name}, config);
+
+  model.fetch()
+    .then(() => {
+      // Save the user name and continue
+      req.sessionModel.set('name', name);
+      return super.validate(req, res, next);
+    })
+    .catch(error => {
+      if (error.status === 404) {
+        // When a user name which is not associated
+        // with an GitHub account is submitted
+        // inform the user with a custom
+        // Validation error
+        error = {
+          'name': new ValidationError('name', {
+            type: 'not-found'
+          })
+        };
+      }
+      // When the error is anything else
+      // let the error bubble up to the errorhandler
+      next(error);
+    });
+}
+```
+
+When the request is valid the next step in the system is invoked. In this case `/show` will be loaded and the associated behaviour invoked.
+If the request does not meet the validation requirements then the error is assigned to the `locals`, compiled and rendered as a validation message.
+If the request fails for a reason other than being unable to find the user, the error is handled by the default error handler and, which in this case will render an error template.
+
+
+### Fetch repos
+The `fetch-repos` behaviour is designed to retrieve the `name` set previously to the session and use it to GET all repositories from the Github API for that user.
+
+The `fetch-repos` beahaviour `require`'s the `fetch-repos` model, a child of the `github` model, which parses the response before returning it to the behaviour.
+```js
+const Model = require('../models/fetch-repos');
+```
+
+It has one method, `getValues`, called on a GET request, which immediately calls upt to `super.getValues` to access the values stored on the previous POST.
+
+```js
+// `getValues` is part of the GET pipeline. For details
+// see https://github.com/UKHomeOfficeForms/hof-behaviour-hooks.
+getValues(req, res, next) {
+
+  // Call super to retrieve the values
+  super.getValues(req, res, (err, values) => {
+    if (err) {
+      // Return early on error
+      return next(err);
+    }
+    // Define an instance of the Model
+    // with attributes and config (uri)
+    const model = new Model({name: values.name}, config);
+
+    // Call `fetch` to get the resource
+    // from the url we configured
+    model.fetch().then(repos => {
+
+      // Augment the values from super
+      // with the repos from the API
+      values.repos = repos;
+
+      // Call back with the values
+      next(null, values);
+    })
+    .catch(err => {
+      // Let the error bubble up to the errorhandler
+      next(err);
+    });
+  });
+
+}
+```
+
+When the request is successful, the repos are assigned to the locals, compiled and rendered in the frontend.
```

Comment:
I'd hope it'd be obvious because I have a step called `show`, which also determines the name of the template. I suppose a small note in `index.js` that mentions the name of the template for that step will be `show`. I'd hope that we will cover the basics in another tutorial though.